recipes/xorg/all/conanfile.py

```diff
@@ -95,3 +95,9 @@ def package_info(self):
                      "xcb-xinerama", "xcb", "xkeyboard-config", "xcb-atom", "xcb-aux", "xcb-event", "xcb-util",
                      "xcb-dri3"]:
             self._fill_cppinfo_from_pkgconfig(name)
+        if self.settings.os == "Linux":
+            if "-luuid" not in self.cpp_info.components["sm"].sharedlinkflags:
+                self.cpp_info.components["sm"].sharedlinkflags.append("-luuid")
+            if "-luuid" not in self.cpp_info.components["sm"].exelinkflags:
+                self.cpp_info.components["sm"].exelinkflags.append("-luuid")
```

Comment:
I'm doing something cagey by just adding the -luuid flag here because on the systems that require it, if libSM is installed, uuid will also be installed so we don't need to provide it ourselves, we can link to it and expect it to be there (in fact I tried this by building on centos and ubuntu, without the yum and deb dependencies, the test package fails to build).
That's because libSM depends on the system provided uuid.

In addition to that, some packages carry a dependency to _libuuid_ (the same thing I believe), but given the stability of the library, I'm not too bothered. In practice I pray that the libuuid I'm vendoring via conan won't break the system provided one.

Adding `requires = "uuid"` would not work as there is no uuid package if I understand correctly.
Adding a dependency to libuuid would add an additional burden to the application packager (why should I install a vendored uuid when the system one works fine? I'm using the system xorg libs after all!)

Reply:
I don't get the problem. `self.cpp_info.components["sm"].requires.append("uuid")` creates a dependency on the `uuid` component of the xorg recipe. this component will be populated by the recipe if we simply add "uuid" to the for loop above.

Reply:
> could we do `self.cpp_info.components["sm"].requires = "uuid"` instead ? So that we transitively get uuid's dependencies too

Were you thinking of something like this (adding uuid to the packages and picking from pkg-config)? That sounds neat
```python
    def package_info(self):
        for name in [ #snipped for brevity 
                    "xcb", "xkeyboard-config", "xcb-atom", "xcb-aux", "xcb-event", "xcb-util",
                     "xcb-dri3", "uuid"]:
            self._fill_cppinfo_from_pkgconfig(name)
        if self.settings.os == "Linux":
            self.cpp_info.components["sm"].depends = "uuid"
```